packages/dev/core/src/AudioV2/soundTools.ts

```diff
@@ -0,0 +1,4 @@
+/** @internal */
+export function _CleanUrl(url: string) {
```

Comment:
when is this needed?

Reply:
The holiday playground at https://playground.babylonjs.com/#KR5CRQ#28 has a sound file url with a `#` in it, which doesn't work with `fetch` or when setting an HTMLAudioElement's source, so I added this function to do the url encoding.

Reply:
wow. this should be rejected by any browser. Hashes don't belong in a URL of an asset. I guess you can simply use `encodeURIComponent` on any URL instead of this and solve other potential problems.

Reply:
`encodeURIComponent` is converting everything in the URI, including the path separators and query parameter symbols (`?` and `&`), which is more than what's needed. To get the `_CleanUrl` function working like I want, I'm having to do this...

```TypeScript
const parts = url.split("?");
const paths = parts[0].split("/");
return paths.map((x) => encodeURIComponent(x)).join("/") + (parts[1] ? "?" + parts[1] : "");
```

Is there a better way?

Reply:
Talked about this offline and we decided to leave it as-is.